source/slang/slang-ir-specialize-resources.cpp

```diff
@@ -164,6 +191,8 @@ struct ResourceOutputSpecializationPass
         // into a combined pass that specializes in both directions and
         // also folds in SSA formation to clean up temporaries.
 
+        lastProcessedFunc = oldFunc;
```

Comment:
> Can you think of an alternative way to achieve your goals?

Sure, there is an alternative (easier to follow) solution I have in mind. 

> Why do we need this lastProcessedFunc mechanism?

The general idea of the bug fix is to add some recursion to our resource specialization. The downside is that since the old approach assumed "the failing function is the callee", the assumption which was made no-longer holds true and we need a new way to track which function failed.

`lastProcessedFunc` was (likely) the most efficient solution, there are other solutions which should be much easier to follow which I can do as well.